github-actions/labels-sync/src/main.ts

```diff
@@ -0,0 +1,84 @@
+import * as core from '@actions/core';
+import {context} from '@actions/github';
+import {Octokit, RestEndpointMethodTypes} from '@octokit/rest';
+import {allLabels, Label} from '../../../ng-dev/pr/common/labels.js';
+import {getAuthTokenFor, ANGULAR_ROBOT, revokeActiveInstallationToken} from '../../utils.js';
+
+/** The type for a Github label returned from the Github API.  */
+type GithubLabel =
+  RestEndpointMethodTypes['issues']['listLabelsForRepo']['response']['data'][number];
+
+/** Synchronize the provided managed labels with the given repository. */
+async function syncLabelsInRepo(github: Octokit, repoName: string, managedLabels: Label[]) {
+  core.startGroup(`Syncing ${repoName}`);
+  /** The current repository name and owner for usage in the Github API. */
+  const repo = {repo: repoName, owner: context.repo.owner};
+
+  core.debug(`Requesting labels`);
+  /** The list of current labels from Github for the repository. */
+  const repoLabels = await github.paginate(github.issues.listLabelsForRepo, context.repo);
+  core.debug(`Retrieved ${repoLabels.length} from Github`);
+
+  // For each label in the list of managed labels, ensure that it exists and is in sync.
+  for (const {description, label} of managedLabels) {
```

Comment:
What should happen with additional per-repo specific labels? Maybe we should have a comment or description somewhere stating that currently these are not cleaned up/removed?

Reply:
Added more comment to expand on it.